src/main/java/io/fabric8/jenkins/openshiftsync/OpenShiftUtils.java

```diff
@@ -419,42 +422,48 @@ public static String getJenkinsURL(OpenShiftClient openShiftClient,
     }
 
     public static String getNamespacefromPodInputs() {
-        String namespace = System
-                .getProperty(Constants.OPENSHIFT_PROJECT_ENV_VAR_NAME);
-        if (namespace != null && namespace.trim().length() > 0)
-            return namespace;
-        File f = new File(Constants.OPENSHIFT_PROJECT_FILE);
-        if (f.exists()) {
-            FileInputStream fis = null;
-            try {
-                fis = new FileInputStream(f);
-                ArrayList<Integer> al = new ArrayList<Integer>();
-                int rawbyte = -1;
-                while ((rawbyte = fis.read()) != -1) {
-                    al.add(rawbyte);
-                }
-
-                byte[] buf = new byte[al.size()];
-                for (int i = 0; i < al.size(); i++) {
-                    buf[i] = (byte) al.get(i).intValue();
-                }
-                namespace = new String(buf);
-                if (namespace != null && namespace.trim().length() > 0)
-                    return namespace;
-
-            } catch (FileNotFoundException e) {
-                logger.log(Level.FINE, "getNamespaceFromPodInputs", e);
-            } catch (IOException e) {
-                logger.log(Level.FINE, "getNamespaceFromPodInputs", e);
-            } finally {
+        if (jenkinsPodNamespaceInitted.compareAndSet(false, true)) {
```

Comment:
this is a race...  two callers come in, one does the compare+set (initted is now true), the other sees the value is already set and returns namespace, before the one that did the compare+set actually set the namespace value.

Is there no way to just set this value once during plugin initialization?  or a static init?

Reply:
or don't even do a compare+set, just check if namespace is null.  it does mean going through the logic potentially more than once, but that's not a big deal.  once all the threads have done it, it should be fine, worst case.

Reply:
(I still think a static init would be best though)

Reply:
I was expecting them to block on the compareAndSet, but yeah, upon further review, it doesn't work that way

that is what I get for trying (relatively) newer java concurrency

yeah I can do it the old fashion way